dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/grpc/GrpcCompositeCodec.java

```diff
@@ -16,123 +16,103 @@
  */
 package org.apache.dubbo.rpc.protocol.tri.h12.grpc;
 
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.common.config.ConfigurationUtils;
+import org.apache.dubbo.common.io.StreamUtils;
+import org.apache.dubbo.common.utils.ArrayUtils;
 import org.apache.dubbo.remoting.http12.exception.DecodeException;
 import org.apache.dubbo.remoting.http12.exception.EncodeException;
 import org.apache.dubbo.remoting.http12.message.HttpMessageCodec;
 import org.apache.dubbo.remoting.http12.message.MediaType;
+import org.apache.dubbo.rpc.model.FrameworkModel;
+import org.apache.dubbo.rpc.model.MethodDescriptor;
+import org.apache.dubbo.rpc.model.PackableMethod;
+import org.apache.dubbo.rpc.model.PackableMethodFactory;
```

Comment:
**Suggestion**: Refactoring in codec classes should be carefully validated to ensure that serialization and deserialization functionalities are intact and optimized. <br> **Code Suggestion**: 
 ```
 + import org.apache.dubbo.common.URL;
 + import org.apache.dubbo.common.config.ConfigurationUtils;
 + public GrpcCompositeCodec(URL url, FrameworkModel frameworkModel, String mediaType) {
 +     this.url = url;
 +     this.frameworkModel = frameworkModel;
 +     this.mediaType = mediaType;
 + }
 + loadPackableMethod(methodDescriptor);
 + // Simplified encode and decode methods leveraging the PackableMethod abstraction
 ```